server/metadata/key_encoder.go

```diff
@@ -40,29 +41,20 @@ type Encoder interface {
 	//	   information i.e. whether the index is pkey, etc. The remaining elements are values for this index.
 	EncodeKey(encodedTable []byte, idx *schema.Index, idxParts []interface{}) (keys.Key, error)
 
-	// DecodeTableName is used to decode the key stored in FDB and extract namespace name, database name and collection name.
-	DecodeTableName(tableName []byte) (string, string, string, bool)
+	// DecodeTableName is used to decode the key stored in FDB and extract namespace name, database name and collection ids.
+	DecodeTableName(tableName []byte) (uint32, uint32, uint32, bool)
 	DecodeIndexName(indexName []byte) uint32
 }
 
-// NewEncoder creates Dictionary encoder to encode keys.
-func NewEncoder(mgr *TenantManager) Encoder {
-	return &DictKeyEncoder{
-		mgr: mgr,
-	}
+// NewEncoder creates Dictionary metaStore to encode keys.
+func NewEncoder() Encoder {
+	return &DictKeyEncoder{}
 }
 
 type DictKeyEncoder struct {
-	mgr *TenantManager
 }
 
 func (d *DictKeyEncoder) EncodeTableName(ns Namespace, db *Database, coll *schema.DefaultCollection) ([]byte, error) {
-	if db == nil {
-		return nil, api.Errorf(api.Code_INVALID_ARGUMENT, "database is missing")
```

Comment:
There are already checks at the callers of this function, for example both runner.GetDatbase, runner.GetCollection have this check.
So something will fail before getting to this function.

Reply:
Then let's document the API accordingly because till now the contract was strict so TableName maps to ns, db, and collection and when we decode we strictly know that we will get these three back but with your change, it can mean anything and any API may choose to just call this and Encode it by passing simply a single value(just namespace or just namespace and database).

Reply:
documented